packages/flutter/lib/src/widgets/binding.dart

```diff
@@ -82,6 +88,22 @@ abstract class WidgetsBinding extends BindingBase implements GestureBinding, Ren
         buildOwner.reassemble(renderViewElement);
       }
     );
+
+    registerStringServiceExtension(
+      name: 'applicationRoute',
+      getter: () => _navigatorState?.currentRoute?.settings?.name,
+      setter: (String route) {
+        _navigatorState?.pushNamed(route);
+      }
+    );
+  }
+
+  NavigatorState get _navigatorState {
+    NavigatorOwner observer = WidgetsBinding.instance._observers.firstWhere(
```

Comment:
I agree that this is not pretty - I'm very open to suggestions here :) I need to get the `NavigatorState` for the current app w/o necessarily having access to a `context`.